### PR TITLE
[AKS] Migrate ACS module to track 2 SDK

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

Migrate ACS module to track 2 SDK. 
WIP! Need exhaustive testing!

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by FumingZhang** [comment]
DO NOT MERGE

**Comment by yonzhan** [comment]
ACS

File: src/azure-cli/azure/cli/command_modules/acs/custom.py
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:3541]: Not all `create_or_update` are changed to `begin_create_or_update` in track 2, only LRO (long running operation) is changed to `begin_xxx_xxx_xxx`, you need check SDK source code as source of truth.

**Comment by jsntcy** [comment]
@FumingZhang, what's the status of the migration? any help from us?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:1477]: Hi, the assignments_clients depends on azure.mgmt.authorization, it seems that this module (Role) has not been migrated to track 2 yet?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:3541]: sure, but it seems that there is no ```create_or_update ``` method in "operations/_managed_clusters_operations.py" in azure-mgmt-containerservice-15.1.0 any more.

**Comment by FumingZhang** [comment]
> @FumingZhang, what's the status of the migration? any help from us?

On the one hand, I am proposing this [PR](https://github.com/Azure/azure-rest-api-specs/pull/14537/files) to modify swagger to fix the strange naming problem in track 2 SDK. On the other hand, some of the live test cases fail with track 1 SDK and I am fixing them ([example](https://github.com/Azure/azure-cli/pull/18168)).

File: src/azure-cli/azure/cli/command_modules/acs/custom.py
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: OK, so keep the code unchanged.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:3553]: It becomes `begin_creat_or_update` now.

**Comment by jsntcy** [comment]
> > @FumingZhang, what's the status of the migration? any help from us?
> 
> On the one hand, I am proposing this [PR](https://github.com/Azure/azure-rest-api-specs/pull/14537/files) to modify swagger to fix the strange naming problem in track 2 SDK. On the other hand, some of the live test cases fail with track 1 SDK and I am fixing them ([example](https://github.com/Azure/azure-cli/pull/18168)).

You can merge this PR first without waiting for the swagger fix as our code freeze date is 6/11 in this sprint. You can use new SDK to fix naming issue next sprint in Azure CLI once the swagger fix is merged.

**Review by andyliuliming — APPROVED** [review]

**Comment by andyliuliming** [comment]
let's organize one bug bash before the code be released.

**Review by jsntcy — APPROVED** [review]

**Pull request merged**
